common/views.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import AnonymousUser
+
+from rest_framework import generics
+
+from .models import (
+    Contact, County, SubCounty, Constituency)
+
+from .serializers import (
+    ContactSerializer, CountySerializer, SubCountySerializer,
+    ConstituencySerializer)
+
+
+class FilterViewMixin(object):
+    def get_queryset(self):
+        user = self.request.user
+        if not isinstance(user, AnonymousUser):
+            if user.is_national:
+                return self.queryset
+            else:
+                if user.is_incharge:
+                    return self.queryset.filter(county=user.county)
+                else:
+                    return []
+        else:
+            return self.queryset
+
+
+class ContactView(generics.ListCreateAPIView):
+    queryset = Contact.objects.all()
+    serializer_class = ContactSerializer
+
+
+class ContactDetailView(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Contact.objects.all()
+    serializer_class = ContactSerializer
+    lookup_field = 'id'
```

Comment:
These `lookup_fields` do not change the default behavior ( lookup by `pk`, which is itself a Django alias of `id` - unless you have a funky model setup ).

I suggest that they all be removed.